index/server/pkg/server/endpoint.go

```diff
@@ -305,6 +305,26 @@ func serveDevfileStarterProjectWithVersion(c *gin.Context) {
 			return
 		}
 
+		// Track event for telemetry. Ignore events from the registry-viewer and DevConsole since those are tracked on the client side. Ignore indirect calls from clients.
+		if enableTelemetry && !util.IsWebClient(c) && !util.IsIndirectCall(c) {
+
+			user := util.GetUser(c)
+			client := util.GetClient(c)
```

Comment:
are you passing in the client name `registry-viewer`?  We are ignoring those because we wanted to avoid tracking duplicate events .   Maybe you can add an exception?  Check if client is `registry-viewer` and `starterProjectName` is not empty?  Not sure if that's unique enough to infer it's a download event

Reply:
@kim-tsao As discussed, for the time being we can assume that the client is most likely the registry viewer if the client name is the generic `devfile-registry`. Other clients using this endpoint will be setting the `Client` header property to change this or using their own telemetry.